### ISSUE TITLE
some cases are unstable

### ISSUE THREAD

**Issue by zhw-zhang** [silent-chen/layout-guidance#16]
Hello, thank you for your excellent work, it inspires me a lot. I encountered a problem during testing and hope to get your help. When I use the following example, I can get satisfactory results on the first run.
`   examples = {"prompt": "A panda sitting next to a fish.",
                "phrases": "panda; fish",
                "bboxes": [[[0.1, 0.2, 0.5, 0.8]], [[0.75, 0.6, 0.95, 0.8]]],
                'save_path': cfg.general.save_path
                }`
![image](https://github.com/silent-chen/layout-guidance/assets/135094935/582f1d1e-006e-4dc6-b8cc-8ac3caa6210e)
But when I run it repeatedly 5-6 times, the result becomes like this.
![example_0](https://github.com/silent-chen/layout-guidance/assets/135094935/2dc2a156-bef7-428c-995c-652557e33190)
In my opinion, using the same seed should not cause different results every time. I checked it all night and found no problems in other cases. (Example: "A hello kitty toy is playing with a purple ball.")

My device is NVIDIA A100.

**Comment by silent-chen** [comment]
Hi, thanks for your interest in our work. I am not sure what is the exact reasons that cause the unstable results but I suggest you to fix all random seed for random, numpy, pytorch and disable the cudnn. You can try with following code: 
```
random.seed(seed)
np.random.seed(seed)
torch.manual_seed(seed)
torch.cuda.manual_seed_all(seed)
torch.backends.cudnn.benchmark = False
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.enabled = False
```

**Comment by zhw-zhang** [comment]
Thank you for your reply. I think the previous problem may be an extremely low-probability event (maybe due to underlying configuration issues). After I fixed all the seeds that can be set, the same result still occurred. But when I change the seed or case, the results will always be stable.

**Comment by dreamer121121** [comment]
> Hi, thanks for your interest in our work. I am not sure what is the exact reasons that cause the unstable results but I suggest you to fix all random seed for random, numpy, pytorch and disable the cudnn. You can try with following code:
> 
> ```
> random.seed(seed)
> np.random.seed(seed)
> torch.manual_seed(seed)
> torch.cuda.manual_seed_all(seed)
> torch.backends.cudnn.benchmark = False
> torch.backends.cudnn.deterministic = True
> torch.backends.cudnn.enabled = False
> ```

I also find the same unstable issue, and i find that the optimization method: 
```
grad_cond = torch.autograd.grad(loss.requires_grad_(True), [latents])[0]
```
will get different outputs every time even the input(loss and latents) is fixed
